scripts/lib-franklin.js

```diff
@@ -0,0 +1,753 @@
+ * @param {Element} block The block element
+ * @returns {object} The block config
+ */
+export function readBlockConfig(block) {
+  const config = {};
+  block.querySelectorAll(':scope > div').forEach((row) => {
+    if (row.children) {
+      const cols = [...row.children];
+      if (cols[1]) {
+        const col = cols[1];
+        const name = toClassName(cols[0].textContent);
+        let value = '';
+        if (col.querySelector('a')) {
+          const as = [...col.querySelectorAll('a')];
+          if (as.length === 1) {
+            value = as[0].href;
+          } else {
+            value = as.map((a) => a.href);
+          }
+        } else if (col.querySelector('img')) {
+          const imgs = [...col.querySelectorAll('img')];
+          if (imgs.length === 1) {
+            value = imgs[0].src;
+          } else {
+            value = imgs.map((img) => img.src);
+          }
+        } else if (col.querySelector('p')) {
+          const ps = [...col.querySelectorAll('p')];
+          if (ps.length === 1) {
+            value = ps[0].textContent;
+          } else {
+            value = ps.map((p) => p.textContent);
+          }
+        } else value = row.children[1].textContent;
+        config[name] = value;
+      }
+    }
+  });
+  return config;
+}
+
+/**
+ * Decorates all sections in a container element.
+ * @param {Element} main The container element
+ */
+export function decorateSections(main) {
+  main.querySelectorAll(':scope > div').forEach((section) => {
+    const wrappers = [];
+    let defaultContent = false;
+    [...section.children].forEach((e) => {
+      if (e.tagName === 'DIV' || !defaultContent) {
+        const wrapper = document.createElement('div');
+        wrappers.push(wrapper);
+        defaultContent = e.tagName !== 'DIV';
+        if (defaultContent) wrapper.classList.add('default-content-wrapper');
+      }
+      wrappers[wrappers.length - 1].append(e);
+    });
+    wrappers.forEach((wrapper) => section.append(wrapper));
+    section.classList.add('section');
+    section.dataset.sectionStatus = 'initialized';
+    section.style.display = 'none';
+
+    /* process section metadata */
+    const sectionMeta = section.querySelector('div.section-metadata');
+    if (sectionMeta) {
+      const meta = readBlockConfig(sectionMeta);
+      Object.keys(meta).forEach((key) => {
+        if (key === 'style') {
+          const styles = meta.style
+            .split(',')
+            .map((style) => toClassName(style.trim()));
+          styles.forEach((style) => section.classList.add(style));
+        } else {
+          section.dataset[toCamelCase(key)] = meta[key];
+        }
+      });
+      sectionMeta.parentNode.remove();
+    }
+  });
+}
+
+/**
+ * Updates all section status in a container element.
+ * @param {Element} main The container element
+ */
+export function updateSectionsStatus(main) {
+  // NOTE: added div.section to pick up sections that are listed deep
```

Comment:
@snowiewdev do you have an example where this comes to play?

Reply:
Yes, can you give us some examples ? I see 2 cases:
1. either the content structure is wrong and should be adjusted to be able to revert
2. it is a valid use case (embed, fragment, table...?) and the boilerplate would need to be adjusted. Ideally you could make a PR there ;-)

Reply:
Hey @trieloff @kptdobe , thanks for taking a look on the code. 

For the use case in `updateSectionsStatus(main)`, for example in home page, I have a `tabs` block with heading like `Making an impact with our customers from the start`. The tabs logic is built originally in the codebase by Adobe team, we reused the logic and put the `marquee` block inside the tab content. When there's like `Section Metadata` tab and another `block` is put inside the section, the original `...main.querySelectorAll(':scope > div.section)` is not able to pick the `marquee block` up as it's nested in one deeper layer, so I adjusted that to `div.section` in order to pick that up. 

Site Link: https://main--helix-website--adobe.hlx.page/home
Google Doc Link: https://docs.google.com/document/d/1JWYx1suRfhfxu_-9LD4bfGb1NhcppipeUiJhdjqnwu0/edit#heading=h.k7jlztrplgdv

Maybe our use case it's a bit complex but I feel like in future, sometimes developers would like to reuse blocks more and allowing nesting the blocks may be more convenient. But the decision on updating the boilerplate is totally up to the Adobe team. For this PR we are adjusting boilerplate in this repo only and making a note, as I'm not sure if someone in future may replace the whole `lib-franklin.js` by direct copy and paste for pulling the latest update, then the fix we applied will be gone and the site may not look like as expected. I think it may be best if we can address the issue in boilerplate. Please let me know if the code in this PR need to be further adjusted, thanks!

Reply:
Attached screenshot for the `tabs` section:


<img width="1509" alt="Screen Shot 2023-07-11 at 10 30 51 AM" src="https://github.com/adobe/helix-website/assets/72127025/2aa7b8dd-156a-42a9-9275-d97874fe47b8">

Reply:
Not sure I fully follow. There is no "nesting" in your content structure, which is good (say no to nesting!).
You have several consecutive sections (tabs) that are controlled by the first one (tab header). I think this is the right content structure.
I would need to look deeper to understand. If the fact that there is a Section Metadata has an influence on the DOM that is not supported properly by the method in lib-franklin, then we need to fix that in the boilerplate. A PR against the boilerplate would be ideal but... you need to demonstrate the problem in a simple matter.

Reply:
If this is only about the testimonial tabs, then I think changing the content is a better way to do this. I've found myself editing it quite a bit and struggled with the structure. I'll create a draft home that has a content structure that I'd find more palpable as an author and then we can see what it takes to adjust the block.